lib/big_query/rollup.ex

```diff
@@ -0,0 +1,61 @@
+defmodule BigQuery.Rollup do
+  import BigQuery.Base.Query
+
+  # a day isn't "complete" until this many seconds after it's over
+  @buffer_seconds 900
+
+  # get a single day of downloads bucketed by hours
+  def hourly_downloads(), do: hourly_downloads(Timex.now)
+  def hourly_downloads(dtim) do
+    now = Timex.now()
+    day = Timex.beginning_of_day(dtim)
+    case completion_state(day, now) do
+      :none ->
+        {[], %{day: day, complete: false, hours_complete: 0}}
+      :partial ->
+        query_hourly_downloads(day) |> set_meta(:complete, false) |> set_meta(:hours_complete, hours_complete(now))
```

Comment:
For logging, it was useful to indicate how many hours were done.  In the mix task output, you'll see something like:

```
Rollup.HourlyDownloads.2018-05-02 incomplete (14/24 hours)
```